src/main/java/be/seeseemelk/mockbukkit/entity/OcelotMock.java

```diff
@@ -0,0 +1,61 @@
+package be.seeseemelk.mockbukkit.entity;
+
+import be.seeseemelk.mockbukkit.ServerMock;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Ocelot;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.UUID;
+
+public class OcelotMock extends AnimalsMock implements Ocelot
+{
+
+	@Deprecated
+	private Ocelot.Type type = Ocelot.Type.WILD_OCELOT;
+	private boolean isTrusting = false;
+
+	/**
+	 * Constructs a new {@link OcelotMock} on the provided {@link ServerMock} with a specified {@link UUID}.
+	 *
+	 * @param server The server to create the entity on.
+	 * @param uuid   The UUID of the entity.
+	 */
+	public OcelotMock(@NotNull ServerMock server, @NotNull UUID uuid)
+	{
+		super(server, uuid);
+	}
+
+	@Override
+	public boolean isTrusting()
+	{
+		return this.isTrusting;
+	}
+
+	@Override
+	public void setTrusting(boolean trust)
+	{
+		this.isTrusting = trust;
+	}
+
+	@Override
+	@Deprecated
+	public @NotNull Type getCatType()
+	{
+		return this.type;
+	}
+
+	@Override
+	@Deprecated
+	public void setCatType(@NotNull Type type)
+	{
+		this.type = type;
```

Comment:
This should rather be
```suggestion
		 throw new UnsupportedOperationException("Cats are a different entity!");
```

Reply:
But why put an unsupported operation, because then people who are going to test with this outdated method won't be able to do so?

Reply:
This is exactly the behavior observed in paper, and as MockBukkit is meant to reflect paper, it should fail as well with the same exception.

Reply:
Oh dear, I hadn't seen that, I'll change it.

Reply:
Done in the commit : [d3cea18](https://github.com/MockBukkit/MockBukkit/pull/868/commits/d3cea1842cdd7e5f07992e3474a02ede3eae4438)

Reply:
Cool. Hmm could you modify the unit tests as well for this? Otherwise, looks good to me

Reply:
This was done here 8669a4539fe1e314411dced3ae4c51961e1a95ba